kmp-library/src/iosMain/kotlin/com/novoda/aws/v4/signer/hash/Sha256Encoder.kt

```diff
@@ -0,0 +1,31 @@
+package com.novoda.aws.v4.signer.hash
+
+import kotlinx.cinterop.*
+import platform.CoreCrypto.CC_SHA256
+import platform.CoreCrypto.CC_SHA256_DIGEST_LENGTH
+import platform.Foundation.NSString
+import platform.Foundation.NSUTF8StringEncoding
+import platform.Foundation.dataUsingEncoding
+import platform.Foundation.dataWithLength
+
+actual object Sha256Encoder {
+
+    @ExperimentalUnsignedTypes
+    actual fun encode(value: String): String {
```

Comment:
This is the kotlin/native implementation of SHA-256 using `platform.CoreCrypto.CC_SHA256`.
I found this implementation in the Kotlin/Native slack, so this looks also for me like black magic ^^